demos/python_demos/whiteboard_inpainting_demo/whiteboard_inpainting_demo.py

```diff
@@ -0,0 +1,198 @@
+
+sys.path.append(osp.join(osp.dirname(osp.dirname(osp.abspath(__file__))), 'common'))
+import monitors
+
+set_log_config()
+WINNAME = 'Whiteboard_inpainting_demo'
+
+
+def expand_mask(detection, d):
+    for i in range(len(detection[0])):
+        detection[0][i][2] = extend_mask(detection[0][i][2], d)
+
+
+def extend_mask(mask, d=70):
+    contours, _ = cv2.findContours(mask, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+    for contour in contours:
+        for c in contour:
+            for x, y in c:
+                cv2.circle(mask, (x, y), d, (1, 1, 1), -1)
+    return mask
+
+
+def remove_background(img, kernel_size=(7, 7), blur_kernel_size=21, invert_colors=True):
+    bgr_planes = cv2.split(img)
+    result = []
+    kernel = np.ones(kernel_size, np.uint8)
+    for plane in bgr_planes:
+        dilated_img = cv2.morphologyEx(plane, cv2.MORPH_OPEN, kernel)
+        dilated_img = cv2.dilate(dilated_img, kernel)
+        bg_img = cv2.medianBlur(dilated_img, blur_kernel_size)
+        if invert_colors:
+            diff_img = 255 - cv2.absdiff(plane, bg_img)
+        else:
+            diff_img = cv2.absdiff(plane, bg_img)
+        result.append(diff_img)
+    return cv2.merge(result)
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Whiteboard inpainting demo')
+    parser.add_argument('-i', type=str, help='Input sources (index of camera \
+                        or path to a video file)', required=True)
+    parser.add_argument('-m_i', '--m_instance_segmentation', type=str, required=False,
+                        help='Path to the instance segmentation model')
+    parser.add_argument('-m_s', '--m_semantic_segmentation', type=str, required=False,
+                        help='Path to the semantic segmentation model')
+    parser.add_argument('-t', '--threshold', type=float, default=0.6,
+                        help='Threshold for person instance segmentation model')
+    parser.add_argument('--output_video', type=str, default='', required=False,
+                        help='Optional. Path to output video')
+    parser.add_argument("--no_show", help="Optional. Don't show output", action='store_true')
+
+    parser.add_argument('-d', '--device', type=str, default='CPU',
+                        help='Optional. Specify a target device to infer on. CPU, GPU, FPGA, HDDL or MYRIAD is '
+                             'acceptable. The demo will look for a suitable plugin for the device specified')
+    parser.add_argument('-l', '--cpu_extension', type=str, default=None,
+                        help='MKLDNN (CPU)-targeted custom layers.Absolute \
+                              path to a shared library with the kernels impl.')
+    parser.add_argument('-u', '--utilization_monitors', default='', type=str,
+                        help='Optional. List of monitors to show initially')
+    args = parser.parse_args()
+
+    if not osp.isfile(args.i) and not str.isdigit(args.i):
```

Comment:
You shouldn't check that the file exists, because OpenCV supports RTSP URIs as inputs, which aren't files. Also, AFAIK, the camera ID needs to actually be an integer for OpenCV to treat it as the camera ID.

The usual approach here it to cast the input to an `int` and if that raises a `ValueError`, keep it as a string.

Reply:
The check removed. Proposed approach is implemented in the `VideoCapture` class.